### ISSUE TITLE
[New Feature] Add new loss functions

### ISSUE THREAD

**Issue by roshan-acharya** [ravin-d-27/PyDeepFlow#93]
The losses.py module currently supports basic loss functions like MSE, BCE, Huber, Categorical Crossentropy, and Hinge loss. To expand its utility for different deep learning tasks, additional loss functions should be added.

Proposed Additions:

L1 Loss (Mean Absolute Error) – robust to outliers, often used in image translation tasks.

Dice Loss – useful for segmentation tasks to measure mask overlap.

Focal Loss – handles class imbalance for classification problems.

KL Divergence Loss – used in probabilistic models and VAEs.

Cosine Similarity Loss – compares embeddings or feature vectors

**Comment by roshan-acharya** [comment]
Assigne it to me

**Comment by roshan-acharya** [comment]
@ravin-d-27 Can you assign this to me
